pkg/model/iam/iam_builder.go

```diff
@@ -49,12 +49,35 @@ const PolicyDefaultVersion = "2012-10-17"
 
 // Policy Struct is a collection of fields that form a valid AWS policy document
 type Policy struct {
-	Statement []*Statement
-	Version   string
+	clusterName         string
+	unconditionalAction sets.String
+	clusterTaggedAction sets.String
+	Statement           []*Statement
+	Version             string
 }
 
 // AsJSON converts the policy document to JSON format (parsable by AWS)
 func (p *Policy) AsJSON() (string, error) {
+	if len(p.unconditionalAction) > 0 {
+		p.Statement = append(p.Statement, &Statement{
+			Effect:   StatementEffectAllow,
+			Action:   stringorslice.Of(p.unconditionalAction.List()...),
+			Resource: stringorslice.String("*"),
+		})
+	}
+	if len(p.clusterTaggedAction) > 0 {
+		p.Statement = append(p.Statement, &Statement{
+			Effect:   StatementEffectAllow,
+			Action:   stringorslice.Of(p.clusterTaggedAction.List()...),
+			Resource: stringorslice.String("*"),
+			Condition: Condition{
+				"StringEquals": map[string]string{
+					"aws:ResourceTag/KubernetesCluster": p.clusterName,
```

Comment:
Do we have a plan to move off of the legacy tags? Perhaps we could add a condition that allows both? (for future PR)

Reply:
I _think_ in-tree CCM still relies on the legacy tags. But I think we can eventually move away from this soon. Putting this into one location anyways make this a lot simpler.